### PR TITLE
check_format: Add fail_on_diff + small cleanups

### PR DESCRIPTION
**Pull request by phlax**
This speeds up the check_format in ci/format/format_pre job by optimizing the fix/check algo and adding a `fail_on_diff` flag

This allows CI to call `--fix` directly without having to call check, and then fix

The algo for checking/fixing itself is also slightly improved

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @mattklein123 re conv in dev channel, this should be a pretty big improvement over what was there previously

i have thinned out the (~non-ci) runs in the format_pre elsewhere so with these changes applied perf should be better in all situations and running more things

**Comment by phlax** [comment]
actually to qualify this it is probs still slower if no failure as it does more things

i know of several fairly immediate optis, so ill follow up

File: ci/format_pre.sh [hunk 1]
```diff
@@ -57,14 +57,8 @@ if [[ -n "$AZP_BRANCH" ]]; then
...
 
-fix_format () {
-    echo "Fixing format..."
-    "${ENVOY_SRCDIR}"/tools/code_format/check_format.py fix
-    return 1
-}
-
 CURRENT=check_format
-"${ENVOY_SRCDIR}"/tools/code_format/check_format.py check || fix_format
+"${ENVOY_SRCDIR}"/tools/code_format/check_format.py fix
```
Comment (mattklein123) [inline ci/format_pre.sh:61]: Don't you need to specify fail on diff here? Can you make a format error and double check that CI fails? Thank you.

/wait-any

Comment (phlax) [reply ci/format_pre.sh]: good spot, apologies, fixed

**Comment by mattklein123** [comment]
It's fine I just want to make sure that it works in CI. If you are sure it's good that's fine with me otherwise a test can't hurt.

**Comment by phlax** [comment]
ill land and monitor ci - im pretty confident all is good

**Pull request merged**